### PR TITLE
Fixup `lib_name` when using `PYO3_CONFIG_FILE`

### PR DESCRIPTION
**Pull request by messense**
This matches the logic in `default_cross_compile`: https://github.com/PyO3/pyo3/blob/eafbbc54170e46c6b16067dabcf849a3864d6198/pyo3-build-config/src/impl_.rs#L1415-L1426

### CODE REVIEW

File: pyo3-build-config/src/impl_.rs [hunk 1]
```diff
@@ -458,19 +477,11 @@ print("mingw", get_platform().startswith("mingw"))
 
     #[allow(clippy::unnecessary_wraps)]
     pub fn fixup_import_libs(&mut self) -> Result<()> {
```
Comment (adamreichold) [inline pyo3-build-config/src/impl_.rs:479]: Isn't the function name a misnomer now as it only generates the import libraries now?

File: pyo3-build-config/src/impl_.rs [hunk 2]
```diff
@@ -440,6 +440,25 @@ print("mingw", get_platform().startswith("mingw"))
         let version = version.ok_or("missing value for version")?;
         let implementation = implementation.unwrap_or(PythonImplementation::CPython);
         let abi3 = abi3.unwrap_or(false);
+        // Fixup lib_name if it's not set
+        let lib_name = lib_name.or_else(|| {
+            if let Ok(Ok(target)) = env::var("TARGET").map(|target| target.parse::<Triple>()) {
+                if target.operating_system == OperatingSystem::Windows {
```
Comment (adamreichold) [inline pyo3-build-config/src/impl_.rs:446]: If this logic is the same as the one linked in the cover letter, would it make sense to factor it out into a helper function?

Comment (messense) [reply pyo3-build-config/src/impl_.rs]: Done.

File: pyo3-build-config/src/impl_.rs (hunk 1 above)
Comment (messense) [reply pyo3-build-config/src/impl_.rs:468]: How about `generate_import_libs`? Or any other suggestions?

Comment (adamreichold) [reply pyo3-build-config/src/impl_.rs:468]: The suggested name sounds fine to me. I wonder if the cfg flags can be simplified, e.g. pub on the outside of the function instead using blocks?

**Review by adamreichold — APPROVED** [review]

**Pull request merged**